packages/xo-web/src/common/xo/index.js

```diff
@@ -525,14 +526,27 @@ subscribeVolumeInfo.forceRefresh = (() => {
 })()
 
 const unhealthyVdiChainsLengthSubscriptionsBySr = {}
-export const createSrUnhealthyVdiChainsLengthSubscription = sr => {
-  sr = resolveId(sr)
-  let subscription = unhealthyVdiChainsLengthSubscriptionsBySr[sr]
+export const subscribeSrUnhealthyVdiChainsLength = (sr, cb) => {
+  const srId = resolveId(sr)
+  let subscription = unhealthyVdiChainsLengthSubscriptionsBySr[srId]
   if (subscription === undefined) {
-    subscription = createSubscription(() => _call('sr.getUnhealthyVdiChainsLength', { sr }))
-    unhealthyVdiChainsLengthSubscriptionsBySr[sr] = subscription
+    subscription = createSubscription(() => getUnhealthyVdiChainsLength(srId))
+    unhealthyVdiChainsLengthSubscriptionsBySr[srId] = subscription
+  }
+  return subscription(cb)
+}
```

Comment:
Why this change? :thinking:

Reply:
to pass a callback, I used it to display the warning in the menu. 
See https://github.com/vatesfr/xen-orchestra/pull/6120/files#diff-b64a5e59a5b86374d5fb231988d92c70c47a5e83dac329aebbc1bc9a6177273aR223